department-of-veterans-affairs/abd-vro
# MCP-2160: Update PDF template mapping

## What was the problem?

The PDF generator could not route the same diagnostic code to multiple template versions. 

Associated tickets or Slack threads:

- [MCP-2160](https://amida.atlassian.net/browse/MCP-2160)

## How does this fix it?[^1]

Adds a new field in the request model to signify versioning of the pdf templates.
The VRO controller defaults to `v1` and Mas route should be set up for `v2`.

How to test this PR
- Step 1 Build and run app, try the pdf generator endpoint with `pdfTemplate` set to `v1` or `v2` for hypertension. Use the pdf fetching endpoint and see that two different version of the pdf are created.
- Step 2[^secrel]


[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
